db/collection_index.go

```diff
@@ -222,7 +223,8 @@ func (c *collection) createIndex(
 	c.indexes = append(c.indexes, colIndex)
 	err = c.indexExistingDocs(ctx, txn, colIndex)
 	if err != nil {
-		return nil, err
+		removeErr := colIndex.RemoveAll(ctx, txn)
```

Comment:
question: Since this operation is happening on a transaction, why do we have to bother removing the indexed documents if it fails? Can we not rely on the discarding of the transaction?

Reply:
I thought about it myself, but wasn't sure if it would be correct to discard the whole transaction. After all, it theoretically might contain several operations (not only index creation) it should be up to the caller's site to decide what to do.
On the other hand, calling `colIndex.RemoveAll` is very localized and the collection has all necessary knowledge to make this decision. 
Maybe I'm missing something.
Note: here is the test that tests it https://github.com/sourcenetwork/defradb/pull/2131/files#diff-0a00fc4743373305afbef72902d95cdc5a9ddbda32c694a0c97c85d03efcde46R1034